Automattic/woocommerce-payments
# Improve code quality of `RecentDepositsList` component, including outdated `Next Deposit` references, complexity/nesting reduction

Fixes #8889

#### Changes proposed in this Pull Request









This is a small PR that removes outdated references to `Next deposit` in the `RecentDepositsList` component. We don't render next/estimated deposits in this table anymore – only deposits with the status: `paid` `pending` `in_transit` `canceled` `failed`.

While I was here I made some minor code-quality improvements:

* A more explicit return `null` instead of `<></>` for the "return nothing" case
* Removed the redundant `JSX.Element` return type
* Inlined the JSX for the deposit rows to reduce unnecessary code complexity/nesting

**Context**
I saw the comments referring to `Next deposit` and had to double-check that server no longer returns any next/estimated deposits, only deposits with the status: `paid` `pending` `in_transit` `canceled` `failed`.

#### Testing instructions





* Observe code changes and confirm this is an improvement to code quality 
* Ensure no references to `Next deposit` remain in this file
* Visit Payments > Overview and ensure the table renders as expected

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/b6c4864f-7510-4c2e-8194-5e4f1c2be17c)


-------------------


**Post merge**
